includes/plugins/plugins.php

```diff
@@ -0,0 +1,104 @@
+<?php
+/**
+ * Utility functions to handle Platform plugins.
+ *
+ * @package hm-platform
+ */
+
+namespace HM\Platform\Plugins;
+
+use HM\Platform;
+use HM\Platform\Config as Config;
+
+/**
+ * Load the other files in the namespace.
+ */
+function setup() {
+	require_once __DIR__ . '/loaders.php';
+}
+
+/**
+ * Get a list of all available plugins.
+ *
+ * @return array Plugins and the associated configuration.
+ */
+function get_available_plugins() {
+	return array_keys( Config\get_config()['plugins'] );
+}
+
+/**
+ * Get a list of all enabled plugins.
+ *
+ * @return array Plugins and the associated configuration.
+ */
+function get_enabled_plugins() {
+	$enabled_plugins = [];
+
+	foreach( Config\get_config()['plugins'] as $plugin => $enabled ) {
+		if ( ! $enabled ) {
+			continue;
+		}
+
+		$enabled_plugins[] = $plugin;
+	}
+
+	return $enabled_plugins;
+
+}
+
+/**
+ * Load all enabled plugins, along with their customisation files.
+ */
+function load_enabled_plugins() {
+	foreach ( get_enabled_plugins() as $name ) {
+		switch ( $name ) {
+			case 'aws_ses_wp_mail':
+				Loaders\load_aws_ses_wp_mail();
+				break;
+
+			case 'cavalcade':
+				Loaders\load_cavalcade();
+				break;
+
+			default:
+				require_once Platform\PLUGIN_DIR . get_plugin_file_path( $name );
+				break;
+		}
+	}
+}
+
+/**
+ * Return the path, relative to the Platform plugin directory, to the main file of a plugin.
+ *
+ * @param string $config_slug Plugin slug, in config format.
+ *
+ * @return string Relative path to the main plugin file.
+ */
+function get_plugin_file_path( string $config_slug ) {
+	// Exceptions that do not respect the naming pattern.
+	switch( $config_slug ) {
+		case 'cavalcade':
```

Comment:
It's pretty unlikely with the variety of plugins we add that they'll always follow naming conventions so there'll be some overhead in keeping this up to date along with a manifest for some metadata and the yaml config tree builder.

A manifest file with an array like you mentioned could help avoid this logic and it's likely we'll need something like that anyway because of localisations for plugin descriptions etc. That's the main problem to solve with this approach that I can see.

Reply:
WordPress plugins _should_ follow naming conventions, and those who don't include an extra file to load the main plugin file.

IMO I can't see why we should use an extra function over something like enforcing `plugin-name/plugin-name.php`. In this case `plugin-name` is a slug, that is kept the same throughout the languages.

Reply:
Our current engineering guidelines advocate the use of `plugin.php` as the main plugin file right now so this isn't something trivial to change to or enforce right here. I'm also keen to avoid having to make such a change to every plugin we add from a maintenance point of view when applying updates from upstream.